tests/integration/repo_merge_upstream_test.go

```diff
@@ -147,5 +147,59 @@ func TestRepoMergeUpstream(t *testing.T) {
 				return queryMergeUpstreamButtonLink(htmlDoc) == ""
 			}, 5*time.Second, 100*time.Millisecond)
 		})
+
+		t.Run("FastForwardOnly", func(t *testing.T) {
```

Comment:
Is it possible to simplify the tests (to save CI time)?

IMO we only need to do 2 tests:

1. ff_only=false: fail
2. ff_only=true: succeed

Other cases should have been covered by other tests.